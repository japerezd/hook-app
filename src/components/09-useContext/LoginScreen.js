import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    
    // 1. obtener referencia al userContext
    const { setUser} = useContext(UserContext)
    // 2. setUser
    // 3. {id:1234, name: 'jorge'}
    const user = {
        id: 1234,
        name: 'Jorge'
    }
    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            <button 
                className="btn btn-primary"
                onClick={() => setUser(user)}
            >Login</button>
        </div>
    )
}
