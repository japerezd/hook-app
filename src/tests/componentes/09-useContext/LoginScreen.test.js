import { mount, shallow } from 'enzyme';
import React from 'react';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Pruebas sobre <LoginScreen />', () => {
    
    const setUser = jest.fn();
    
    // Debe hacerse con mount
    const wrapper = shallow(
        <UserContext.Provider value = { { setUser } } >

            <LoginScreen />
        </UserContext.Provider>
    );
    test('debe mostrarse correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    });


    test('debe ejecutarse el setUser con el argumento esperado', () => {
        // wrapper.find('button').simulate('click'); se puede hacer con esto tambien
        wrapper.find('button').prop('onClick')();

        expect(setUser).toHaveBeenCalledWith({
            name: 'Beto',
            email: 'beto@gmail.com'
        });

    });
    
    
});
