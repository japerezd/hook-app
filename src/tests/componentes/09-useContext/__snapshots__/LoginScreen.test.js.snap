// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <LoginScreen /> debe mostrarse correctamente  1`] = `<LoginScreen />`;
